### PR TITLE
feat: Add any links associated with a table to querybook descriptions

### PR DESCRIPTION
**Pull request by zhangvi7**
### Description
- Tables from metatore maybe have links associated to table descriptions
- Exposing any links that are attached to tables in Querybook
- Adding `table_links` column to the data_table_information table and load it from metastore.
### Test Plan
image

### CODE REVIEW

File: querybook/migrations/versions/767cf0f573bd_add_table_links_to_datatable.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+"""add table links to datatable
```
Comment (kgopal492) [inline querybook/migrations/versions/767cf0f573bd_add_table_links_to_datatable.py:1]: You should probably edit this file to only include the changes made in this PR (aka add column table_links for upgrade and drop_column table_links for downgrade)

File: querybook/webapp/const/metastore.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ export interface IDataTable {
         partition_keys?: string[];
     };
     custom_properties?: Record<string, string | number>;
-
+    table_links?: Record<string, string>[];
```
Comment (jczhong84) [inline querybook/webapp/const/metastore.ts:77]: it looks like a link can only have `label` and `url`, can we define a new type for it with `label` and `url`, while `label` as optional?

File: querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx [hunk 3]
```diff
@@ -130,6 +130,16 @@ export const DataTableViewOverview: React.FC<
         />
     ) : null;
 
+    const tableLinks = table.table_links ?? [];
+    const tableLinksDOM = tableLinks.map((link, index) => (
```
Comment (jczhong84) [inline querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx:134]: nit: seems unnecessary to introduce a new variable, just `(table.table_links ?? []).map`

File: querybook/server/const/metastore.py [hunk 4]
```diff
@@ -75,6 +75,9 @@ class DataTable(NamedTuple):
     # Custom properties
     custom_properties: dict[str, str] = None
 
+    # Links associated to table
+    table_links: list[dict[str, str]] = None
```
Comment (czgu) [inline querybook/server/const/metastore.py:79]: could we use typed dict here?

File: querybook/webapp/const/metastore.ts (hunk 2 above)
Comment (czgu) [reply querybook/webapp/const/metastore.ts:77]: +1, using Record is not clear

**Comment by zhangvi7** [comment]
- fixed lint
- addressed comments

**Review by czgu — APPROVED** [review]

**Pull request merged**